client/src/components/buttons/DialogButton.js

```diff
@@ -0,0 +1,66 @@
+"use strict";
+
+import React, { useState } from "react";
+import Button from '@material-ui/core/Button';
+import TextField from '@material-ui/core/TextField';
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogContentText from '@material-ui/core/DialogContentText';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import DateFnsUtils from "@date-io/date-fns";
+import { DatePicker, MuiPickersUtilsProvider } from "@material-ui/pickers";
+
+
+export default function FormDialog(props) {
```

Comment:
It could be better to use classes - So we can use component lifecycle methods later on

Reply:
Agreed. Tried to separate the code into two classes (DialogButton, AddMealDialog) to have at least the button class reusable but for some reason I wasnt able to separate the logic between the classes, and the onClick function did not work. I renamed the class to avoid confusion (since it is not only a button), but maybe one of you can have a look and help me out here to separate the classes.